### ISSUE TITLE
How to best loop over Helm subcharts or a set of manifests

### ISSUE THREAD

**Issue by AndersBennedsgaard** [helm/helm#10171]
TLDR: is it possible to deploy an arbitrary number of the same sub-chart with an umbrella chart?

I am working with Flux to use Git as a source of truth for the contents on our clusters, with multi-tenancy very similar to what is set up in https://github.com/fluxcd/flux2-multi-tenancy, and looking for help on the best solution to a problem I've run into.

We're working with the case where a single tenant could want an arbitrary number of projects to run, with other tenants not being able to access their projects.
To make life simpler, I chose to create a Helm chart that would set up RBAC, Istio rules, monitoring, and syncing with an repository where a tenant can place YAML that should be applied to the cluster.
My solution was to create a list/array of projects in the main chart, making `values.yaml` look something like the following:
```
# Git sync settings
git:
  url: ""
  ref:
    branch: master
  ...

# Project definitions
projects:
  []
  ## The name of the project
  # - name: test-project
  ## The path on Git to what should be applied to the cluster
  #   gitPath: "./test-project/dev"
  #   rbac:
  #     serviceAccount:
  #       name: ""
  #     kind: RoleBinding
  #     roleRef:
  #       kind: ClusterRole
  #       name: tenant
  #   ...

# Project default values. Overridden by projects.*
defaults:
  rbac:
    kind: RoleBinding
    roleRef:
      kind: ClusterRole
      name: tenant
  ...
```

This chart sets up an admin namespace for that tenant, together with one namespace for each of the projects developed by the tenant. 
My solution creates a lot of problems with scopes and default project values (especially with boolean values because of https://github.com/helm/helm/issues/3308), which made me consider creating a sub chart that would set up every project in itself. 

This would make `values.yaml` into something along the line of what is found in [how-do-i-create-an-arbitrary-number-of-instances-of-the-same-dependency-chart-in](https://stackoverflow.com/questions/60770559/how-do-i-create-an-arbitrary-number-of-instances-of-the-same-dependency-chart-in) and [loop-over-helm-subcharts-or-a-set-of-manifests](https://stackoverflow.com/questions/68028198/loop-over-helm-subcharts-or-a-set-of-manifests):
```
projects:
 - name: projectA
   gitPath: "./projectA/dev"
   rbac: 
     ...
 - name: projectB
   gitPath: "./projectB/dev"
```
I have not tried much to try this, since I'm not quite sure where to begin, which is why I am asking you for help :smiley:

**Comment by joejulian** [comment]
If I'm understanding the use case correctly, I would use this to create flux HelmReleases and Kustomization resources and let the Flux controllers handle the rest.

**Comment by AndersBennedsgaard** [comment]
> If I'm understanding the use case correctly, I would use this to create flux HelmReleases and Kustomization resources and let the Flux controllers handle the rest.

Yes exactly. That's what I meant by "... syncing with an repository where a tenant can place YAML that should be applied to the cluster.", but your way of putting it is better ;)
